conformance/utils/suite/features.go

```diff
@@ -64,6 +64,9 @@ const (
 	//
 	//       See: https://github.com/kubernetes-sigs/gateway-api/issues/1780
 	SupportGatewayClassObservedGenerationBump SupportedFeature = "GatewayClassObservedGenerationBump"
+
+	// This option indicates support for Gateway to support dynamic ports in their listeners
+	SupportGatewayListenerDynamicPorts SupportedFeature = "GatewayListenerDynamicPorts"
```

Comment:
Thinking about this in terms of the future when a GatewayClass publishes supported features in status, this particular feature name doesn't really provide much information. IE an implementation could potentially claim this if they supported ports 80, 443, and 8080 if they configured only those ports when running this test. That could be a completely separate set of ports compared to another implementations that also claimed to support this feature.

I'm not sure what the solution is here, just concerned that this doesn't really provide a good way for implementations to actually signal what they support.

/cc @mlavacca @shaneutt

Reply:
> What would be useful is if a GatewayClass publishes the dynamic range of ports it supports

I think we're saying the same thing here, I also want to be able to represent this in GWC status. I'm just coming from the perspective of an implementation that does not support a contiguous set of ports, so we'd need a way to represent that.

Reply:
Seems like that would warrant a GEP no? and is out of scope for this PR?

Reply:
I'm not sure. The work @shaneutt and @mlavacca are doing with Conformance Profiles is leading to a point where conformance tests will be run based on something like `supportedFeatures` populated in GatewayClass status. So far that's the only input we need for all existing conformance tests. This is the first conformance test I'm aware of that also _requires_ additional inputs, and if we're adding it without a plan to migrate to the GWC status-based approach, we're just creating a headache for ourselves in the future.

Reply:
Ok make sense. 

I guess something else to discuss - do we feel the port ranges should appear on the GatewayClass Status or would it be better on the Gateway?